### ISSUE TITLE
Custom Group Keeps Disappearing

### ISSUE THREAD

**Issue by Singers30** [b-morgan/Skillet-Classic#175]
I have a custom 'Grouping' that I named "Prior Cata". I have 4 groups here with various recipes.
One I name "Epic Cata Mail" and put the new Cataclysm ILVL 359 recipes under it. 
Every single time I log back into this character, this group is gone. Deleted itself somehow. 
I've tried logging out, exiting the game while in Stormwind, ALT F4'ing, everything. Group is always gone when I come back. 
Am I doing something wrong here? My Alchemy groups don't delete themselves.

**Comment by b-morgan** [comment]
This is going to get a bit complicated.

First, I need to see your global saved variables file which is saved to disk when you logout. Create the group that disappears and then logout. Make a copy of the `Skillet-Classic.lua` file in `...\World of Warcraft\_classic_\WTF\Account\<your name>\SavedVariables`. Now login, verify that the group has disappeared, and logout again. Make another copy of the saved variables file (with a new name). Now make a copy of the player specific saved variables file `Skillet-Classic.lua` in `...\World of Warcraft\_classic_\WTF\Account\<your name>\<realm name>\<character name>\SavedVariables` (with another new name).

Put all three files in a .zip file and upload it here. I'll probably need you to enable debug output and I may need to provide some custom code. I'll let you know after I look at the above.

**Comment by b-morgan** [comment]
This is going to get a bit complicated.

First, I need to see your global saved variables file which is saved to disk when you logout. Create the group that disappears and then logout. Make a copy of the Skillet-Classic.lua file in `...\World of Warcraft\_classic_\WTF\Account\<your name>\SavedVariables`. Now login, verify that the group has disappeared, and logout again. Make another copy of the saved variables file (with a new name). Now make a copy of the player specific saved variables file in `...\World of Warcraft\_classic_\WTF\Account\<your name>\<realm name>\<character name>\SavedVariables`

**Comment by Singers30** [comment]
[Skillet-Classic(LogoutWithVanishingGroup).zip](https://github.com/user-attachments/files/15744654/Skillet-Classic.LogoutWithVanishingGroup.zip)
[Skillet-Classic(LogoutAfterGroupVanished).zip](https://github.com/user-attachments/files/15744655/Skillet-Classic.LogoutAfterGroupVanished.zip)

Here is what you requested. The issue is happening with Singorz, the hunter and her Leatherworking. The "Mail" group is the group that is present in the first file, and gone in the second.

**Comment by b-morgan** [comment]
I think I see the problem but I haven't figured out what causes it (or how to fix it) so I'd like to perform an experiment.

After logging out with the vanishing group in place, I'd like you to edit the global saved variable file. From the file you sent me find in the `["profiles"] = {`, `		["Singorz - Atiesh"] = {`,   

```
			["groupDB"] = {
				["2108:Cata"] = {
					["Cata Hunter Self Enchants"] = "4:'Draconic Embossment - Agility:'Dragonbone Leg Reinforcements",
					["*ALL*"] = "33:g5:g2:g4:g5",
					["Mail"] = "5:'Dragonkiller Tunic:'Twilight Scale Chestguard:'Corded Viper Belt:'Stormleather Sash",
					["Cata Enchants"] = "5:'Charscale Leg Armor:'Dragonscale Leg Armor:'Drakehide Leg Armor:'Twilight Leg Armor:'Scorched Leg Armor",
					["Leather"] = "2:'Chestguard of Nature`s Fury",
				},
```
and change the `["*ALL*"]` and the `["Mail"]` lines as follows:
```
			["groupDB"] = {
				["2108:Cata"] = {
					["Cata Hunter Self Enchants"] = "4:'Draconic Embossment - Agility:'Dragonbone Leg Reinforcements",
					["*ALL*"] = "33:g5:g2:g4:g6",
					["Mail"] = "6:'Dragonkiller Tunic:'Twilight Scale Chestguard:'Corded Viper Belt:'Stormleather Sash",
					["Cata Enchants"] = "5:'Charscale Leg Armor:'Dragonscale Leg Armor:'Drakehide Leg Armor:'Twilight Leg Armor:'Scorched Leg Armor",
					["Leather"] = "2:'Chestguard of Nature`s Fury",
				},
```
and login again. Let me know if the symptoms change.

**Comment by b-morgan** [comment]
BTW, I didn't write this code, I inherited it. It is probably the most complicated feature of Skillet-Classic (and Skillet for retail). Have patience as we work through this together.

**Comment by Singers30** [comment]
Okay I did what you said and after logging back in, the group was there still for the first time.

**Comment by b-morgan** [comment]
Fantastic! So now I know what the problem is. Somehow, when you created the new group it received a duplicate number. We fixed it manually. I believe you may have deleted a group in the past (because group 3 was missing) and I don't know if that contributed to the problem. 

I'll leave this issue open and do some experiments but the crisis is averted for now. Thanks for helping.

**Comment by Faheetah** [comment]
Just to add on, the same thing happened to me except the groups looked normal but I was getting the following error. I don't know why the database didn't show any duplicates but removing the groups and recreating them fixed it. I think I  might have accidentally dragged a group as a subgroup but no clue how that didn't show in the config. I readded entries by hand in the config but noticed group 7 got duplicated when I created another group. It's fixed but just leaving this here just in case

Date: 2025-03-17 08:24:16
ID: 1
Error occured in: Global
Count: 1
Message: Interface/AddOns/Skillet-Classic/RecipeGroups.lua line 465:
   stack overflow
Debug:
   [string "@Interface/AddOns/Skillet-Classic/RecipeGroups.lua"]:465: RecipeGroupFlatten()
   [string "@Interface/AddOns/Skillet-Classic/RecipeGroups.lua"]:479: RecipeGroupFlatten()
   [string "@Interface/AddOns/Skillet-Classic/RecipeGroups.lua"]:479: RecipeGroupFlatten()

The entirety of the group config:

["3908:Custom"] = {
["Dreamweave"] = "5:'Dreamweave Circlet:'Dreamweave Gloves:'Dreamweave Vest",
["*ALL*"] = "7:g2:g5:g4:g7:g3:g6",
["Felcloth"] = "4:'Felcloth Boots:'Felcloth Hood:'Felcloth Pants:'Felcloth Robe:'Felcloth Shoulders",
["Bags"] = "2:'Felcloth Bag:'Mooncloth Bag:'Runecloth Bag:'Soul Pouch",
["Mooncloth"] = "3:'Mooncloth Leggings:'Mooncloth Robe:'Mooncloth Shoulders:'Mooncloth Vest",
["Shadoweave"] = "6:'Shadoweave Boots:'Shadoweave Gloves:'Shadoweave Mask:'Shadoweave Pants:'Shadoweave Robe:'Shadoweave Shoulders",
}

**Comment by b-morgan** [comment]
@Faheetah,
The original error was a group "disappearing". You claim "the same thing happened to me" but you posted an error. I don't consider this a "same thing".

I can't begin to attempt a fix for your issue without before and after saved variable data. If you can reproduce the problem, please open a new issue and provide the necessary data.
